test/integration/spec/localparticipant.js

```diff
@@ -428,20 +429,21 @@ describe('LocalParticipant', function() {
         if (when === 'previously') {
           thisParticipant.unpublishTrack(thisTrack);
 
-          await Promise.all(thoseParticipants.map(thatParticipant => {
+          await waitFor(thoseParticipants.map(thatParticipant => {
             return tracksUnpublished(thatParticipant, thisParticipant._tracks.size);
-          }));
+          }), 'tracksUnpublished');
         }
 
-        [thisLocalTrackPublication, thoseTracksPublished, thoseTracksSubscribed] = await Promise.all([
-          thisParticipant.publishTrack.apply(thisParticipant, priority ? [thisTrack, { priority }] : [thisTrack]),
-          ...['trackPublished', 'trackSubscribed'].map(event => {
-            return Promise.all(thoseParticipants.map(async thatParticipant => {
-              const [trackOrPublication] = await waitForTracks(event, thatParticipant, 1);
-              return trackOrPublication;
-            }));
-          })
-        ]);
+        const thisLocalTrackPublicationPromise = priority ? thisParticipant.publishTrack(thisTrack, { priority }) : thisParticipant.publishTrack(thisTrack);
+        const thoseTracksPublishedPromise = thoseParticipants.map(thatParticipant => waitForTracks('trackPublished', thatParticipant, 1));
+        const thoseTracksSubscribedPromise = thoseParticipants.map(thatParticipant => waitForTracks('trackSubscribed', thatParticipant, 1));
+
+        thisLocalTrackPublication = await waitFor(thisLocalTrackPublicationPromise, 'local track to publish');
+        thoseTracksPublished = await waitFor(thoseTracksPublishedPromise, 'participants to receive trackPublished');
+        thoseTracksSubscribed = await waitFor(thoseTracksSubscribedPromise, 'participants to receive trackSubscribed');
```

Comment:
@makarandp0 How about:
```js
[thisLocalTrackPublication, thoseTracksPublished, thoseTracksSubscribed] = await Promise.all([
  waitFor(thisLocalTrackPublicationPromise, 'local track to publish'),
  waitFor(thoseTracksPublishedPromise, 'participants to receive trackPublished'),
  waitFor(thoseTracksSubscribedPromise, 'participants to receive trackSubscribed')
]);
```
This way, we can retain the parallelism in the code before you made the changes.

Reply:
@manjeshbhargav - The parallelism is still there, since promises are kicked off in parallel. We are waiting for them in fulfil in sequence, but that won't hurt parallelism, since overall it takes same time.